### DISCUSSION TITLE
JS files won't work

### DISCUSSION THREAD

**Discussion by alexwachowski** [community/community#108536] [New to GitHub]
I've been trying to work on and upload as many projects as I can recently but for some reason none of my .js files will work so anything that I've made with react won't work. (or really any javascript I write that isn't inline). 

here's an example of what I'm talking about. 

here is the repository for my code: https://github.com/alexwachowski/testing

and here is the what currently displays after a url is created: https://alexwachowski.github.io/testing/

I'm not sure what I'm missing so any help is appreciated. 

(I mentioned this in a previous question but I'm posting again for visibility.)

**Comment by spenserblack** [comment]
If you open up the dev console in your browser, you should see an error
```
Uncaught TypeError: Failed to resolve module specifier "three". Relative references must start with either "/", "./", or "../".
```

I noticed in your source code that you're referencing `vite.svg` (another missing asset). Is your project supposed to be built with Vite?

**Reply by alexwachowski** [reply]
Does that mean that the three.js library isn't loading? How do I include the three.js library so that anyone can view it? I'm new so please excuse what might be easy questions. 

It was built with Vite but the svg is just an asset from a template I forgot to get rid of.

**Reply by spenserblack** [reply]
I *think* what you've deployed is actually the source code. What you'll want to deploy is the output of `vite build`. Vite actually has docs specifically for deploying to GitHub pages, which might help you out :slightly_smiling_face: 

https://vitejs.dev/guide/static-deploy.html#github-pages

**Reply by alexwachowski** [reply]
I'll check this out but I should mention I've been having the exact same issue with a blog project. https://github.com/alexwachowski/my-blog
This one is strictly html, css, and js

**Reply by spenserblack** [reply]
I wouldn't say it's the same issue. In *that* project, you're rendering the README.md as the homepage. I don't have much insight into how you set up your deployment there, but I can see from using inspect element at https://alexwachowski.github.io/my-blog/ that it's getting rendered by Jekyll. This is what GitHub Pages uses to create pages from Markdown.

I also see in that project that you're using express.js. That's server-side code. GitHub Pages isn't going host your server for you -- it will just serve *client-side* files, that may or may not be built (building with Vite or Jekyll for example). So, typically, you'd be serving
- HTML
- CSS
- *Client-side* (i.e. frontend) JavaScript -- What runs in the browser, *not* what runs by calling `node`

**Comment by SamuelParrales** [comment]
I checked your index.html https://github.com/alexwachowski/testing/blob/main/index.html
![image](https://github.com/community/community/assets/68031754/5ea7ef16-6576-4c23-8736-e73df1f2902a)

I found that you link **/index.js** when it should be **index.js**, because **/index.js** consider the url this way https://alexwachowski.github.io/index.js while **index.js** https://alexwachowski.github.io/testing/index.js

If you want you can check my index.html: https://github.com/SamuelParrales/app-nobody-answers-you-no/blob/main/index.html
demo: https://samuelparrales.github.io/app-nobody-answers-you-no/

**Reply by alexwachowski** [reply]
This didn't seem to work. I should specify too that I'm having the exact same issue with another project. https://github.com/alexwachowski/my-blog
Someone mentioned that there could be an issue that I used Vite with the three.js project but for this blog it's strictly html, css, and js
